src/main/java/xbot/common/subsystems/drive/control_logic/HeadingAssistModule.java

```diff
@@ -52,31 +71,44 @@ public HeadingAssistModule(
      * @return values between -1 and 1, in terms of "how hard you should try in order to satisfy rotational goals."
      */
     public double calculateHeadingPower(double humanRotationalPower) {
-        
+
         // if human rotational power above some threshold, return that.
         // Also, update a timestamp that says this happened recently
         if (humanRotationalPower > humanThreshold.get()) {
             inAutomaticMode = false;
             lastHumanInput = Timer.getFPGATimestamp();
             return humanRotationalPower;
         }
-        
+
         // If not under threshold, but too close to timestamp,
         // "coast"
         double timeSinceHumanInput = Timer.getFPGATimestamp() - lastHumanInput;
-        
+
         if (timeSinceHumanInput < coastTime.get()) {
             return 0;
         } 
         else if (timeSinceHumanInput >= coastTime.get() && !inAutomaticMode) {
             desiredHeading = pose.getCurrentHeading().getValue();
             inAutomaticMode = true;
             headingModule.reset();
+            decayModule.reset();
             return 0;
         }
         // by this point, the only option left is that the machine is in automatic mode.
         else {
-            return headingModule.calculateHeadingPower(desiredHeading);
+            if (headingMode == null) {
+                log.warn("Heading module did not have any mode set. Forcing to HoldOrientation.");
+                this.headingMode = HeadingAssistMode.HoldOrientation;
+            }
+
+            switch (headingMode) {
+                case HoldOrientation:
+                    return headingModule.calculateHeadingPower(desiredHeading);
+                case DecayVelocity:
+                    return decayModule.calculateHeadingPower(0);
```

Comment:
I still don't understand how this works. This function takes a desired absolute heading. If you call it with 0, won't it just rotate to heading 0? And in the OP, you said it specifically was designed to use only D in the controller. How is that done?

Reply:
the DecayModule's PID parameters are set outside of this class - for an example, see HeadingAssistModuleTest:

```
HeadingModule hold = clf.createHeadingModule(pf.createPIDManager("Hold", 1000, 0, 0));
        HeadingModule decay = clf.createHeadingModule(pf.createPIDManager("Decay", 0, 0, 1000));
        clf.createHeadingAssistModule(hold, decay);
```

A PID controller with only D will just resist changes in error, regardless of the target.

Reply:
Ah, I get it; I completely missed that those two parameters are assisted rather than injected. Can you add some Javadoc on the `HeadingModule` and `HeadingAssistModule` to explain what they do and how they're different?

Reply:
Great idea! Will do.